### DISCUSSION TITLE
Python masters, help me set up the environment...

### DISCUSSION THREAD

**Discussion by freemegu** [community/community#45241] [Programming Help]
### Select Topic Area

Product Feedback

### Body

When using it as Python vscode,
How do you set the data type to come out like that when you over or drag the mouse?
I think everyone can use it on basic, but I can't.
Python version 3.11.1, vscode 1.74.3.
I ask for help from the masters.
![캡처](https://user-images.githubusercontent.com/123458669/214346661-c4b23f03-7a49-4e62-9641-363b4d385557.PNG)

https://user-images.githubusercontent.com/123458669/214456651-6938780c-184b-4697-99d2-1478620b11da.mp4

**Comment by VladislavDev** [comment]
Hi!

You can use the Python Type Hints extension. To do this, go to the Extensions section (on the left in the form of four squares) and enter approximately "python type" in the search.
![image](https://user-images.githubusercontent.com/123298615/214349204-19625283-177c-4c7a-812b-27bdf04847d9.png)
![image](https://user-images.githubusercontent.com/123298615/214349325-d5322da0-9c3b-4e2e-b58a-ca5c22eba122.png)

Click Install and use it when installed.

Good luck!

**Reply by freemegu** [reply]
Thank you very much for your response.
However, I am looking for a way to automatically open the window when I click or drag the mouse.
Please refer to the clip, thanks again.

https://user-images.githubusercontent.com/123458669/214352877-8c0cac07-4407-4e0f-96ca-688b73a6d04e.mp4

**Reply by jge162** [reply]
On a Mac, you can also enable the "auto-display" of the data type by settings, you can do this by:

Open your settings in VS Code by going to File > Preferences > Settings or by pressing Ctrl+, (Cmd+, on Mac).
Search for "python.hoverShowDocsOnSingleClick" and set it to true.
Save your settings and restart VS Code.

**Reply by freemegu** [reply]
Thanks for your reply, but i cant find that setting item "python.hoverShowDocsOnSingleClick" on vscode setting on a window

**Comment by freemegu** [comment]
네이버 메일 앱에서 보냈습니다.
날짜: 2023.01.25 오전 09:47:32 (GMT+09:00)
제목: Re: [community/community] Python masters, help me set up the environment... (Discussion #45241)

@freemegu ok found it, its in the json file, go to the top of vscode where it says search. type >preferences: open Settings (JSON). then control f search for variablesType and set to true. Mine is false. see pic below

**Reply by jge162** [reply]
I posted that earlier but didn’t think that was what u wanted. As it’s static.

**Reply by freemegu** [reply]
ok, got it. it's just simple problem, but very hard to solve :(

**Reply by jge162** [reply]
It’s in settings too. Under python analysis. Enable inlays. But hover is missing. Has to be a bug.

**Reply by freemegu** [reply]
I think so, too. I was too obsessed. 
I should download it again.
Thank you for your help.

**Reply by jge162** [reply]
> I think so, too. I was too obsessed.
> I should download it again.
> Thank you for your help.

U should just use pycharm instead. Well I prefer pycharm.

**Reply by jge162** [reply]
> I think so, too. I was too obsessed.
> I should download it again.
> Thank you for your help.

I use vscode for JavaScript

**Reply by freemegu** [reply]
All online courses use vscode..
I'll take this opportunity to change it.

**Reply by jge162** [reply]
That is true. Everyone trains in vscode. And codespace is vscode.

**Reply by freemegu** [reply]
Your words were very helpful.
Thank you very much.

**Comment by cehfisher** [comment]
Thanks for posting in the GitHub Community, @freemegu! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've asked @queenofcorgis to move this issue into the correct category for you. Good luck!
